src/__swaps__/screens/Swap/components/SwapBottomPanel.tsx

```diff
@@ -102,7 +132,7 @@ export function SwapBottomPanel() {
               onPressWorklet={() => {
                 'worklet';
                 if (type.value !== 'hold') {
-                  SwapNavigation.handleSwapAction();
+                  runOnJS(handleHwConnectionAndSwap)();
```

Comment:
I don't think this is gonna change the performance anyway but have a question, 
will `SwapNavigation.handleSwapAction` that is now inside `runOnJS(handleHwConnectionAndSwap)` run on the js thread? I think so right? previously it ran in the ui thread

Reply:
Yeah. That's a good point. I think you're correct, but it's worth keeping an eye on.